deployment/uberspace-ansible/deploy.yml

```diff
@@ -0,0 +1,114 @@
+---
+- hosts: servers
+  vars:
+    install_root: ~/
+    project_repo: https://github.com/c3foc/hagrid.git
+  gather_facts: false
+
+  tasks:
+
+    - name: copy django_secret generator
+      template:
+        src: files/generate_django_secret.j2
+        dest: "{{ install_root }}generate_django_secret.sh"
+    - name: create django_secret file
+      shell: "sh {{ install_root }}generate_django_secret.sh"
+    - name: Read django_secret
+      shell: "cat {{ install_root }}.django-secret"
+      register: cmd_output
+    - name: Set django_secret as fact
+      set_fact:
+        django_secret: "{{ cmd_output.stdout }}"
+
+    - name: uberspace add domain
+      shell: uberspace web domain add {{ url }}
+      ignore_errors: yes  # if domain is added already, there is a non-zero exit code
+    - name: uberspace set application web backend
+      shell: uberspace web backend set {{ url }}/ --http --port 8000
+    - name: uberspace set static web backend
+      shell: uberspace web backend set {{ url }}/static/ --apache
+
+      # as we don't have private media files, we can serve them via apache
+    - name: uberspace set media web backend
+      shell: uberspace web backend set {{ url }}/media/ --apache
+
```

Comment:
I don't have a clue what the uberspace command does but you may also want to enable caching on the media files within your web server configuration file.

Reply:
uberspace provides an apache proxy. This way we tell it to directly serve media and static files without going through our application server.